content/en/blog/2022/rust-instrumentation/index.md

```diff
@@ -0,0 +1,396 @@
+    error!(it = "accepts multiple arguments", and_also = ("multiple types"), how_many="?", could_be_more_than=some_variable);
+}
+```
+
+Subscriber - For spans and events to be recorded and aggregated we need to initialize a Subscriber. In most cases, we will use an existing Subscriber implementation.
+
+The crate [tracing_subscriber](https://docs.rs/tracing-subscriber/0.3.15/tracing_subscriber/) provides us with general use case subscriber that can be extended and modified according to specific needs.
+
+We initialize the subscriber using Registry struct:
+
+```shell
+let subscriber = tracing_subscriber::Registry::default()
+```
+
+This struct implements a subscriber and exposes us to another important feature of this crate - The Layer. Using layers, we can configure our subscriber to apply specific behaviors when interacting with spans and events.
+
+For example, if we want to filter some of our data, export it, format it or edit it, we can create or use an existing layer and compose it with the registry subscriber.
+
+```shell
+let subscriber = tracing_subscriber::Registry::default().with(SomeLayer)
+```
+
+We will see more examples on how to use these crates to instrument a Rust program in the practical section.
+
+### OpenTelemetry in Rust
+
+The crate that provides support for OpenTelemetry is called - how convenient - [OpenTelemetry](https://docs.rs/opentelemetry/latest/opentelemetry/).
+
+
+Let’s go over some of its key APIs:
+
+- Tracer. Inside the crate, there is a module called traces. It introduces us to a trait called Tracer. This is what tracks and connects our spans to create traces. To use Tracer, we create a subscriber layer. This is how the subscriber knows where to send the spans and how to generate traces.
+
+- Global. This module provides an API so that no matter in what section of the code we currently are, we can access the subscriber, tracer, and propagation context.
+
+- SDK. This module provides an implementation for common use cases with OpenTelemetry. For example, it provides an exporter to send traces to stdout. It also implements context propagation methods that we can use instead of implementing ourselves.
+
+The git repository called [opentelemetry-rust](https://github.com/open-telemetry/opentelemetry-rust) contains implementations of several crates that expand the OpenTelemetry ecosystem. There you can find common instrumentations, exporters, and subscribers use them in a Rust program.
+
+### Distributed tracing in Rust
+
+Using crates from the [opentelemetry-rust](https://github.com/open-telemetry/opentelemetry-rust) repo, we can instrument our application across multiple services and send them to a distributed platform.
+
+For example, we can use the [opentelemetry-jaeger](https://crates.io/crates/opentelemetry-jaeger) to send our traces to a remote Jaeger platform.
+
+Another example is the [opentelemetry-otlp](https://crates.io/crates/opentelemetry-otlp) crate.
```

Comment:
please remove opentelemetry-jaeger and use opentelemetry-otlp, jaeger recommends switching to OTLP (cc [breedx-splk](https://github.com/breedx-splk))